### ISSUE TITLE
Use and standardize .SLP for serialized labels format

### ISSUE THREAD

**Issue by talmo** [murthylab/sleap#304]
Using the .slp extension for HDF5 serialized `sleap.Labels` projects (and predictions) would make it less ambiguous, especially since `tf.keras.Model`s also get serialized to .h5 files (with some annoying hard coded dependencies in their loading code).

This would be analogous to MATLAB .mat files which are just stored in HDF5 format after v7.3.

In a sense, this is like saying that it _is_ an HDF5 file format, but .slp implies a specific set of groups/datasets/attributes, and discourages users from attempting to read the files themselves with h5py.

This also removes the pressure of trying to serialize to a labels serialization format can be read without SLEAP (for now).

Moving forward, we should encourage users to always save projects as .slp files in stages:

1. Initially by just having both the `.slp; .h5` file extension filters in the Save dialog, where users can still save as `.h5` but not by default. The Open file dialog would have a file filter like (`*.slp, *.h5`) so that both formats appear in the file browser by default -- this will avoid confusion for existing users who have all their data saved with `.h5` extensions.

2. Remove the `.h5` option from the dropdown in the Save dialog altogether, so users can no longer save with `.h5` extension from the GUI. The Open file dialog would still have both by default (`*.slp, *.h5`).

3. Bump the `.h5` option in the Open file dialog to the dropdown, so that the default filter is just `*.slp`.

4. Remove the `.h5` option from the Open file dialog altogether, and maybe move it over to the "Import" file browser, so that old labels can still be opened but are effectively "imported" from an old format.

**Comment by talmo** [comment]
I think we have everything except step 4, but I think it's alright as is.
